src/app/public/modules/overlay/overlay.component.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ComponentFactoryResolver,
+  ComponentRef,
+  ElementRef,
+  Injector,
+  OnDestroy,
+  StaticProvider,
+  Type,
+  ViewChild,
+  ViewContainerRef
+} from '@angular/core';
+
+import {
+  NavigationStart,
+  Router
+} from '@angular/router';
+
+import {
+  fromEvent,
+  Observable,
+  Subject
+} from 'rxjs';
+
+import {
+  takeUntil
+} from 'rxjs/operators';
+
+import {
+  SkyOverlayConfig
+} from './overlay-config';
+
+import {
+  SkyOverlayInstance
+} from './overlay-instance';
+
+/**
+ * @internal
+ */
+@Component({
+  selector: 'sky-overlay',
+  templateUrl: './overlay.component.html',
+  styleUrls: ['./overlay.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyOverlayComponent implements OnDestroy {
+
+  public get closed(): Observable<void> {
+    return this._closed;
```

Comment:
Why is the private `_closed` property necessary?

Reply:
We don't want to return the actual `Subject` to the end user because they would be able to trigger their own subscriptions. If we only return an `Observable`, the public method `next` is not available to them. However, I did forget to call `asObservable()` on the subject. I'll add that now.